<h1>Содержание<span class="tocSkip"></span></h1>
<div class="toc"><ul class="toc-item"><li><span><a href="#Функции-и-библиотеки" data-toc-modified-id="Функции-и-библиотеки-1"><span class="toc-item-num">1&nbsp;&nbsp;</span>Функции и библиотеки</a></span></li><li><span><a href="#Обзор-данных" data-toc-modified-id="Обзор-данных-2"><span class="toc-item-num">2&nbsp;&nbsp;</span>Обзор данных</a></span></li><li><span><a href="#Предобработка-данных" data-toc-modified-id="Предобработка-данных-3"><span class="toc-item-num">3&nbsp;&nbsp;</span>Предобработка данных</a></span><ul class="toc-item"><li><span><a href="#устранение-некачественных-данных" data-toc-modified-id="устранение-некачественных-данных-3.1"><span class="toc-item-num">3.1&nbsp;&nbsp;</span>устранение некачественных данных</a></span></li><li><span><a href="#изменение-индексов" data-toc-modified-id="изменение-индексов-3.2"><span class="toc-item-num">3.2&nbsp;&nbsp;</span>изменение индексов</a></span></li></ul></li><li><span><a href="#Разделение-исходных-данные-на-обучающую-и-валидационную-выборки,-подготовка-признаков" data-toc-modified-id="Разделение-исходных-данные-на-обучающую-и-валидационную-выборки,-подготовка-признаков-4"><span class="toc-item-num">4&nbsp;&nbsp;</span>Разделение исходных данные на обучающую и валидационную выборки, подготовка признаков</a></span><ul class="toc-item"><li><span><a href="#локация-1" data-toc-modified-id="локация-1-4.1"><span class="toc-item-num">4.1&nbsp;&nbsp;</span>локация 1</a></span></li><li><span><a href="#локация-2" data-toc-modified-id="локация-2-4.2"><span class="toc-item-num">4.2&nbsp;&nbsp;</span>локация 2</a></span></li><li><span><a href="#локация-3" data-toc-modified-id="локация-3-4.3"><span class="toc-item-num">4.3&nbsp;&nbsp;</span>локация 3</a></span></li></ul></li><li><span><a href="#Обучение-модели" data-toc-modified-id="Обучение-модели-5"><span class="toc-item-num">5&nbsp;&nbsp;</span>Обучение модели</a></span><ul class="toc-item"><li><span><a href="#локация-1" data-toc-modified-id="локация-1-5.1"><span class="toc-item-num">5.1&nbsp;&nbsp;</span>локация 1</a></span></li><li><span><a href="#локация-2" data-toc-modified-id="локация-2-5.2"><span class="toc-item-num">5.2&nbsp;&nbsp;</span>локация 2</a></span></li><li><span><a href="#локация-3" data-toc-modified-id="локация-3-5.3"><span class="toc-item-num">5.3&nbsp;&nbsp;</span>локация 3</a></span></li></ul></li><li><span><a href="#Расчет-прибыли" data-toc-modified-id="Расчет-прибыли-6"><span class="toc-item-num">6&nbsp;&nbsp;</span>Расчет прибыли</a></span><ul class="toc-item"><li><span><a href="#расчет-достаточного-объема-сырья-для-безубыточной-разработки-новой-скважины" data-toc-modified-id="расчет-достаточного-объема-сырья-для-безубыточной-разработки-новой-скважины-6.1"><span class="toc-item-num">6.1&nbsp;&nbsp;</span>расчет достаточного объема сырья для безубыточной разработки новой скважины</a></span></li><li><span><a href="#отбор-скважин-и-расчет-прибыли" data-toc-modified-id="отбор-скважин-и-расчет-прибыли-6.2"><span class="toc-item-num">6.2&nbsp;&nbsp;</span>отбор скважин и расчет прибыли</a></span></li><li><span><a href="#расчет-рисков-и-прибыли" data-toc-modified-id="расчет-рисков-и-прибыли-6.3"><span class="toc-item-num">6.3&nbsp;&nbsp;</span>расчет рисков и прибыли</a></span><ul class="toc-item"><li><span><a href="#локация-1" data-toc-modified-id="локация-1-6.3.1"><span class="toc-item-num">6.3.1&nbsp;&nbsp;</span>локация 1</a></span></li><li><span><a href="#локация-2" data-toc-modified-id="локация-2-6.3.2"><span class="toc-item-num">6.3.2&nbsp;&nbsp;</span>локация 2</a></span></li><li><span><a href="#локация-3" data-toc-modified-id="локация-3-6.3.3"><span class="toc-item-num">6.3.3&nbsp;&nbsp;</span>локация 3</a></span></li></ul></li></ul></li><li><span><a href="#Заключение" data-toc-modified-id="Заключение-7"><span class="toc-item-num">7&nbsp;&nbsp;</span>Заключение</a></span></li></ul></div>

# Выбор локации для скважины

**Входные данные** —  пробы нефти в трёх регионах (характеристики для каждой скважины в регионе уже известны). 



**Дальнейшее использование результатов исследования** — определение наиболее прибыльных мест бурения скважин.

**Цель исследования** 
- построение модели для определения региона, где добыча принесёт наибольшую прибыль, в т.ч.:    
- анализ возможной прибыли и рисков, в т.ч.:
    - расчет достаточного объёма сырья для безубыточной разработки новой скважины
    - нахождение распределение прибыли
    - нахождние средней прибыли, доверительный интервал и риск убытков
- предложение региона для разработки

**Дополнительные условия**
- Для обучения модели подходит только линейная регрессия.
- При разведке региона исследуют 500 точек, из которых с помощью машинного обучения выбирают 200 лучших для разработки.
- Бюджет на разработку скважин в регионе — 10 млрд рублей.
- Один баррель сырья приносит 450 рублей дохода, доход с каждой единицы продукта составляет 450 тыс. руб. (объём указан в тысячах баррелей).
- Риск-аппетит по риску убытки составляет 2.5%.

## Функции и библиотеки

In [46]:
import pandas as pd

import numpy as np

from sklearn.preprocessing import StandardScaler 

from sklearn.model_selection import train_test_split

from sklearn.linear_model import LinearRegression

from sklearn.metrics import mean_squared_error

from statistics import mean

from scipy import stats as st

from sklearn.model_selection import cross_val_score

from sklearn.metrics import fbeta_score, make_scorer

In [2]:
# назначение - расчет доходов полученых при добыче в скважинах, отобранных согласно предсказаниям модели
# входные данные:
# target - набор реальных значений доступных для добычи объемов продукции
# probabilities - набор предсказанных объемов продукции
# count - количество скважин из которых предполагается осуществлять добычу (при значении 10 выбирается 10 скважин с наибольшими предсказанными запасами)

def revenue(target, probabilities, count):
    probs_sorted = probabilities.sort_values(ascending=False) # сортировка предсказанных объемов продукции по возрастанию
    selected = target[probs_sorted.index][:count] # выбор реальных объемов, содержащихся в заданом количестве скважинах, наиболее перспективных согласно предсказаниям   
    
    
    return round(450000 * selected.sum() - 10000000000,0) #расчет реального дохода полученного при разработке скважин (выбранных обученной моделью)

## Обзор данных

Данные геологоразведки трёх регионов:

Признаки:
- id — уникальный идентификатор скважины;
- f0, f1, f2 — значимые признаки;

**Целевой признак**
- product — объём запасов в скважине (тыс. баррелей).

In [3]:
try:
    df_1 = pd.read_csv ('D:/Data for projects/ML/geo_data_0.csv')
    df_2 = pd.read_csv ('D:/Data for projects/ML/geo_data_1.csv')
    df_3 = pd.read_csv ('D:/Data for projects/ML/geo_data_2.csv')

except: #метод try/except применяется для открытия блокнота по адресу Практикума
    df_1 = pd.read_csv ('/datasets/geo_data_0.csv')
    df_2 = pd.read_csv ('/datasets/geo_data_1.csv')
    df_3 = pd.read_csv ('/datasets/geo_data_2.csv')

In [4]:
print (df_1.info())
print()
print (df_2.info())
print()
print (df_3.info())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 100000 entries, 0 to 99999
Data columns (total 5 columns):
id         100000 non-null object
f0         100000 non-null float64
f1         100000 non-null float64
f2         100000 non-null float64
product    100000 non-null float64
dtypes: float64(4), object(1)
memory usage: 3.8+ MB
None

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 100000 entries, 0 to 99999
Data columns (total 5 columns):
id         100000 non-null object
f0         100000 non-null float64
f1         100000 non-null float64
f2         100000 non-null float64
product    100000 non-null float64
dtypes: float64(4), object(1)
memory usage: 3.8+ MB
None

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 100000 entries, 0 to 99999
Data columns (total 5 columns):
id         100000 non-null object
f0         100000 non-null float64
f1         100000 non-null float64
f2         100000 non-null float64
product    100000 non-null float64
dtypes: float64(4), object(1)
memor

In [5]:
print (df_1.nunique())
print()
print (df_2.nunique())
print()
print (df_3.nunique())

id          99990
f0         100000
f1         100000
f2         100000
product    100000
dtype: int64

id          99996
f0         100000
f1         100000
f2         100000
product        12
dtype: int64

id          99996
f0         100000
f1         100000
f2         100000
product    100000
dtype: int64


In [6]:
print (df_1.duplicated().sum())
print()
print (df_2.duplicated().sum())
print()
print (df_3.duplicated().sum())

0

0

0


In [7]:
print (df_1['id'].duplicated().sum())
print()
print (df_2['id'].duplicated().sum())
print()
print (df_3['id'].duplicated().sum())

10

4

4


<div style="border:solid blue 2px; padding: 20px">

По результатам обзора данных выявлено:
    
    - наличие различных комбинаций признаков для одного и того же уникального идентификатора скважины (количество некачественных данных статистически не значимо)

## Предобработка данных

### устранение некачественных данных

В ходе обзора данных выявлено сопоставления различных комбинаций признаков для одних и тех же скважин, восстановить данные не представляется возможным.
Объем некачественных данных статистически не значим и позволяет провести обучение без их предобработки, тем не менее устраним не качественные данные

In [8]:
df_1 = df_1[df_1['id'].duplicated() == False]

In [9]:
df_2 = df_2[df_2['id'].duplicated() == False]

In [10]:
df_3 = df_3[df_3['id'].duplicated() == False]

### изменение индексов

In [11]:
df_1 = df_1.set_index('id')
df_2 = df_2.set_index('id')
df_3 = df_3.set_index('id')

## Разделение исходных данные на обучающую и валидационную выборки, подготовка признаков

### локация 1

In [12]:
features_1 = df_1.drop('product',axis=1) # признаки 
target_1 = df_1['product'] # целевой признак

In [13]:
features_train_1, features_valid_1, target_train_1, target_valid_1 = train_test_split(
    features_1, target_1, test_size=0.25, random_state=404) #выделение обучающей и валидационной выборки

In [14]:
print(features_train_1.shape)
print(features_valid_1.shape)

print(target_train_1.shape)
print(target_valid_1.shape)

(74992, 3)
(24998, 3)
(74992,)
(24998,)


Проведем масштабирование признаков

In [15]:
pd.options.mode.chained_assignment = None

numeric = features_train_1.columns

scaler = StandardScaler()
scaler.fit(features_train_1[numeric])

features_train_1[numeric] = scaler.transform(features_train_1[numeric])
features_valid_1[numeric] = scaler.transform(features_valid_1[numeric])

### локация 2

In [16]:
features_2 = df_2.drop('product',axis=1) # признаки 
target_2 = df_2['product'] # целевой признак

In [17]:
features_train_2, features_valid_2, target_train_2, target_valid_2 = train_test_split(
    features_2, target_2, test_size=0.25, random_state=404) #выделение обучающей и валидационной выборки

In [18]:
print(features_train_2.shape)
print(features_valid_2.shape)

print(target_train_2.shape)
print(target_valid_2.shape)

(74997, 3)
(24999, 3)
(74997,)
(24999,)


Проведем масштабирование признаков

In [19]:
pd.options.mode.chained_assignment = None

numeric = features_train_2.columns

scaler = StandardScaler()
scaler.fit(features_train_2[numeric])

features_train_2[numeric] = scaler.transform(features_train_2[numeric])
features_valid_2[numeric] = scaler.transform(features_valid_2[numeric])

### локация 3

In [20]:
features_3 = df_3.drop('product',axis=1) # признаки 
target_3 = df_3['product'] # целевой признак

In [21]:
features_train_3, features_valid_3, target_train_3, target_valid_3 = train_test_split(
    features_3, target_3, test_size=0.25, random_state=404) #выделение обучающей и валидационной выборки

In [22]:
print(features_train_3.shape)
print(features_valid_3.shape)

print(target_train_3.shape)
print(target_valid_3.shape)

(74997, 3)
(24999, 3)
(74997,)
(24999,)


Проведем масштабирование признаков

In [23]:
pd.options.mode.chained_assignment = None

numeric = features_train_3.columns

scaler = StandardScaler()
scaler.fit(features_train_3[numeric])

features_train_3[numeric] = scaler.transform(features_train_3[numeric])
features_valid_3[numeric] = scaler.transform(features_valid_3[numeric])

## Обучение модели

### локация 1

In [24]:
model_1 = LinearRegression() 
model_1.fit(features_train_1, target_train_1) # обучение
predictions_valid_1 = model_1.predict(features_valid_1) #предсказание

reserves_mean_1 = predictions_valid_1.mean()
RMSE_1 = mean_squared_error (target_valid_1, predictions_valid_1)**0.5 #значение метрики RMSE на валидационной выборке

print('Средний запас предсказанного сырья:',reserves_mean_1)
print("RMSE модели линейной регрессии на валидационной выборке:", RMSE_1)

Средний запас предсказанного сырья: 92.51106814077136
RMSE модели линейной регрессии на валидационной выборке: 37.489657515705076


In [25]:
ftwo_scorer = make_scorer(mean_squared_error)

scores = cross_val_score(model_1, features_1, target_1, cv=5,scoring=ftwo_scorer) 

final_score = scores.mean()**0.5

print('Средняя оценка качества модели:', final_score)

Средняя оценка качества модели: 37.694934636954144


### локация 2

In [26]:
model_2 = LinearRegression() 
model_2.fit(features_train_2, target_train_2) # обучение
predictions_valid_2 = model_2.predict(features_valid_2) #предсказание

reserves_mean_2 = predictions_valid_2.mean()
RMSE_2 = mean_squared_error (target_valid_2, predictions_valid_2)**0.5 #значение метрики RMSE на валидационной выборке

print('Средний запас предсказанного сырья:',reserves_mean_2)
print("RMSE модели линейной регрессии на валидационной выборке:", RMSE_2)

Средний запас предсказанного сырья: 68.93255484171301
RMSE модели линейной регрессии на валидационной выборке: 0.8894296342214345


In [27]:
scores = cross_val_score(model_2, features_2, target_2, cv=5,scoring=ftwo_scorer) 

final_score = scores.mean()**0.5

print('Средняя оценка качества модели:', final_score)

Средняя оценка качества модели: 0.8904258590202471


### локация 3

In [28]:
model_3 = LinearRegression() 
model_3.fit(features_train_3, target_train_3) # обучение
predictions_valid_3 = model_3.predict(features_valid_3) #предсказание

reserves_mean_3 = predictions_valid_3.mean()
RMSE_3 = mean_squared_error (target_valid_3, predictions_valid_3)**0.5 #значение метрики RMSE на валидационной выборке

print('Средний запас предсказанного сырья:',reserves_mean_3)
print("RMSE модели линейной регрессии на валидационной выборке:", RMSE_3)

Средний запас предсказанного сырья: 95.04719299953348
RMSE модели линейной регрессии на валидационной выборке: 40.10447873875969


In [29]:
scores = cross_val_score(model_3, features_3, target_3, cv=5,scoring=ftwo_scorer) 

final_score = scores.mean()**0.5

print('Средняя оценка качества модели:', final_score)

Средняя оценка качества модели: 40.05698887609784


<div style="border:solid blue 2px; padding: 20px">

В результате обучения моделей обнаружены значительные отличия в метрики RMSE между моделями, обученными на данных для различних регионов
кратный рост метрики RMSE соответствует увеличению средних запасов предсказанного сырья

## Расчет прибыли

### расчет достаточного объема сырья для безубыточной разработки новой скважины

минимально необходимый объем добычи в регионе для окупаемости проекта

In [30]:
cost = 10000000000 #сумма затрат на проект
price = 450000 #цена за тыс. ед. продукции

volume_min = cost/price
volume_min #количество тыс.ед. продукции необходимое для покрытия затрат на проект

22222.222222222223

In [31]:
print('Средний запас предсказанного сырья в локации 1:',reserves_mean_1)
print('Средний запас предсказанного сырья в локации 2:',reserves_mean_2)
print('Средний запас предсказанного сырья в локации 3:',reserves_mean_3)

Средний запас предсказанного сырья в локации 1: 92.51106814077136
Средний запас предсказанного сырья в локации 2: 68.93255484171301
Средний запас предсказанного сырья в локации 3: 95.04719299953348


<div style="border:solid blue 2px; padding: 20px">

Все локации удовлетворяют требованию превышения совокупного объема запасов минимально необходимого для достижения окупаемости проекта

минимально необходимый объем добычи для окупаемости проекта (в расчете на одну скважину)

In [32]:
cost = 10000000000 #сумма затрат на проект
price = 450000 #цена за тыс. ед. продукции
n_well = 200 #количество планируемых к разработке скважин

volume_well_min = (cost/price)/n_well
volume_well_min

111.11111111111111

In [33]:
print ('Средний запас предсказанного сырья 200 лучших скважин в локации 1:',mean(sorted(predictions_valid_1, reverse=True)[:199]))
print ('Средний запас предсказанного сырья 200 лучших скважин в локации 2:',mean(sorted(predictions_valid_2, reverse=True)[:199]))
print ('Средний запас предсказанного сырья 200 лучших скважин в локации 3:',mean(sorted(predictions_valid_3, reverse=True)[:199]))

Средний запас предсказанного сырья 200 лучших скважин в локации 1: 155.43210222575053
Средний запас предсказанного сырья 200 лучших скважин в локации 2: 138.74181037950063
Средний запас предсказанного сырья 200 лучших скважин в локации 3: 149.3790255334628


<div style="border:solid blue 2px; padding: 20px">

Все локации удовлетворяют требованию превышения совокупного объема запасов минимально необходимого для достижения окупаемости проекта 

### отбор скважин и расчет прибыли

In [34]:
predictions_valid_1_ = pd.Series(predictions_valid_1)
target_valid_1_ = pd.Series(target_valid_1).reset_index(drop=True)

print ('прибыль от выбранных моделью 200 скважин в регионе 1:',
       revenue(target_valid_1_, predictions_valid_1_, 200))
print ('масимальная прибыль от 200 скважин в регионе 1:',
       round(target_valid_1.sort_values(ascending=False).head(200).sum()*450000,0))

прибыль от выбранных моделью 200 скважин в регионе 1: 3137955218.0
масимальная прибыль от 200 скважин в регионе 1: 16504153163.0


In [35]:
predictions_valid_2_ = pd.Series(predictions_valid_2)
target_valid_2_ = pd.Series(target_valid_2).reset_index(drop=True)

print ('прибыль от выбранных моделью 200 скважин в регионе 2:',
       revenue(target_valid_2_, predictions_valid_2_, 200))
print ('масимальная прибыль от 200 скважин в регионе 2:',
       round(target_valid_2.sort_values(ascending=False).head(200).sum()*450000,0))

прибыль от выбранных моделью 200 скважин в регионе 2: 2415086697.0
масимальная прибыль от 200 скважин в регионе 2: 12415086697.0


In [36]:
predictions_valid_3_ = pd.Series(predictions_valid_3)
target_valid_3_ = pd.Series(target_valid_3).reset_index(drop=True)

print ('прибыль от выбранных моделью 200 скважин в регионе 3:',
       revenue(target_valid_3_, predictions_valid_3_, 200))
print ('масимальная прибыль от 200 скважин в регионе 3:',
       round(target_valid_3.sort_values(ascending=False).head(200).sum()*450000,0))

прибыль от выбранных моделью 200 скважин в регионе 3: 2235954020.0
масимальная прибыль от 200 скважин в регионе 3: 16930573580.0


### расчет рисков и прибыли

#### локация 1

In [37]:
state = np.random.RandomState(12345)
    
values = []
count_losses = 0

for i in range(1000):
    
    target_subsample = target_valid_1_.sample(n=500, replace=True, random_state=state)
    probs_subsample = predictions_valid_1_[target_subsample.index] 
          
    values.append(revenue(target_subsample, probs_subsample, 200))
    
    if revenue(target_subsample, probs_subsample, 200) < 0:
        count_losses += 1
        

values = pd.Series(values)
lower = values.quantile(0.975)
upper = values.quantile(0.025)

mean_1 = values.mean()
print("Средняя прибыль:", mean_1)
print("2,5%-квантиль:", lower)
print("97,5%-квантиль:", upper)

Средняя прибыль: 435174701.893
2,5%-квантиль: 976016201.4749999
97,5%-квантиль: -111512939.64999998


In [38]:
confidence_interval = st.t.interval(alpha=0.95,df=(len(values)-1), loc=mean_1, scale=values.sem())

print("95%-ый доверительный интервал:", confidence_interval)

95%-ый доверительный интервал: (418114579.91965115, 452234823.86634886)


In [39]:
print('риск убытков:',round((count_losses/1000)*100,2),'%')


риск убытков: 5.6 %


#### локация 2

In [40]:
values = []
count_losses = 0

for i in range(1000):
    
    target_subsample = target_valid_2_.sample(n=500, replace=True, random_state=state)
    probs_subsample = predictions_valid_2_[target_subsample.index] 
          
    values.append(revenue(target_subsample, probs_subsample, 200))
    
    if revenue(target_subsample, probs_subsample, 200) < 0:
        count_losses += 1
        

values = pd.Series(values)
lower = values.quantile(0.975)
upper = values.quantile(0.025)

mean_2 = values.mean()
print("Средняя прибыль:", mean_2)
print("2,5%-квантиль:", lower)
print("97,5%-квантиль:", upper)

Средняя прибыль: 483323029.216
2,5%-квантиль: 899506901.7249999
97,5%-квантиль: 90217098.22500001


In [41]:
confidence_interval = st.t.interval(alpha=0.95,df=(len(values)-1), loc=mean_2, scale=values.sem())

print("95%-ый доверительный интервал:", confidence_interval)

95%-ый доверительный интервал: (470377496.9149201, 496268561.51707995)


In [42]:
print('риск убытков:',round((count_losses/1000)*100,2),'%')

риск убытков: 0.9 %


#### локация 3

In [43]:
values = []
count_losses = 0

for i in range(1000):
    
    target_subsample = target_valid_3_.sample(n=500, replace=True, random_state=state)
    probs_subsample = predictions_valid_3_[target_subsample.index] 
          
    values.append(revenue(target_subsample, probs_subsample, 200))
    
    if revenue(target_subsample, probs_subsample, 200) < 0:
        count_losses += 1
        

values = pd.Series(values)
lower = values.quantile(0.975)
upper = values.quantile(0.025)

mean_3 = values.mean()
print("Средняя прибыль:", mean_3)
print("2,5%-квантиль:", lower)
print("97,5%-квантиль:", upper)

Средняя прибыль: 387794301.28
2,5%-квантиль: 907101430.15
97,5%-квантиль: -169779585.975


In [44]:
confidence_interval = st.t.interval(alpha=0.95,df=(len(values)-1), loc=mean_3, scale=values.sem())

print("95%-ый доверительный интервал:", confidence_interval)

95%-ый доверительный интервал: (370880271.8563468, 404708330.70365316)


In [45]:
print('риск убытков:',round((count_losses/1000)*100,2),'%')

риск убытков: 8.0 %


## Заключение

В рамках настоящего исследования:

- проведена предобработка данных
- проведено обучение моделей (LinearRegression) для трех регионов 
- произведен рассчет прибыли в разрезе регионов, включающий в себя:
    - определение объема сырья, необходимого для достижения точки безубыточности проекта
    - отбор скважин с наимбольшими запасами (согласно предсказаниям модели) и расчет прибыли (согласно реальным данным о запасах в выбранных скважинах)
- при помощи техники Bootstrap определны распределения прибыли по регионам, определена средняя прибыль, доверительный интервал и риски убытков, так:
    - для локации 1:
        - прибыль от выбранных моделью 200 скважин в регионе 1: 3 137 955 218.0
        - масимальная прибыль от 200 скважин в регионе 1: 16 504 153 163.0
        - Средняя прибыль: 435 174 701.893
        - риск убытков: 5.6 %

    - для локации 2:
        - прибыль от выбранных моделью 200 скважин в регионе 22 415 086 697.0
        - масимальная прибыль от 200 скважин в регионе 212 415 086 697.0
        - Средняя прибыль: 483 323 029.216
        - риск убытков: 0.9 %

    - для локации 3:
        - прибыль от выбранных моделью 200 скважин в регионе 32 235 954 020.0
        - масимальная прибыль от 200 скважин в регионе 316 930 573 580.0
        - Средняя прибыль: 387 794 301.28
        - риск убытков: 8.0 %
        
        
В рамках исследования рассмотрены три региона, так:
- разработка реиона 1 является менее перспективной, однако расчитанная средняя прибыль для данного региона сопоставима с средней прибылью для региона 2.
- разработка реиона 2 является наиболее прогнозируемой и сопряжена с меньшими рисками. Так же для данного региона расчитана наибольшая средняя прибыль.
- разработка региона 3 допустима, однако сопряжена с большим уровнем риска. Так же стоит отметить, что не смотря на большие запасы 200 наиболее перспективных скважин, расчетная средняя прибыль для данного региона меньше чем для региона 2.

**вывод**

при принятии решения о длительном проекте (в отсутствии в регионе конкурирующих организаций) наиболее перспективным регионом согласно расчетам является регион 2.

при принятии решения о реализации проекта на части скважин (например в случае наличия конкурентов в регионе) и при возможности выбора наиболее перспективных скважин возможна организация добычи в регионе 3.